### PR TITLE
Enable nullability in DataGridViewColumn EventArgs

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in DataGridViewColumn EventArgs.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/4995)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewColumnStateChangedEventArgs.cs [hunk 1]
```diff
@@ -2,19 +2,17 @@
...
 
-#nullable disable
-
 namespace System.Windows.Forms
 {
     public class DataGridViewColumnStateChangedEventArgs : EventArgs
     {
-        public DataGridViewColumnStateChangedEventArgs(DataGridViewColumn dataGridViewColumn, DataGridViewElementStates stateChanged)
+        public DataGridViewColumnStateChangedEventArgs(DataGridViewColumn? dataGridViewColumn, DataGridViewElementStates stateChanged)
         {
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewColumnStateChangedEventArgs.cs:10]:
```suggestion
        public DataGridViewColumnStateChangedEventArgs(DataGridViewColumn dataGridViewColumn, DataGridViewElementStates stateChanged)
        {
            if (dataGridViewColumn is null)
            {
                throw new ArgumentNullException(nameof(dataGridViewColumn));
            }
```

**Review by RussKie — CHANGES_REQUESTED** [review]

**Comment by dreddy-work** [comment]
@gpetrou , checking back if you can take this to closure.

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewColumnStateChangedEventArgs.cs [hunk 2]
```diff
@@ -2,19 +2,22 @@
...
+            {
+                throw new ArgumentNullException(nameof(dataGridViewColumn));
+            }
+
             Column = dataGridViewColumn;
             StateChanged = stateChanged;
         }
 
-        public DataGridViewColumn Column { get; }
+        public DataGridViewColumn? Column { get; }
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewColumnStateChangedEventArgs.cs:20]: changes in this file seems contradicting. If column is expected be null then we shouldn't throw exception.

**Review by dreddy-work — APPROVED** [review]

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewColumnStateChangedEventArgs.cs [hunk 3]
```diff
@@ -2,14 +2,17 @@
...
 namespace System.Windows.Forms
 {
     public class DataGridViewColumnStateChangedEventArgs : EventArgs
     {
         public DataGridViewColumnStateChangedEventArgs(DataGridViewColumn dataGridViewColumn, DataGridViewElementStates stateChanged)
         {
+            if (dataGridViewColumn is null)
+            {
+                throw new ArgumentNullException(nameof(dataGridViewColumn));
+            }
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewColumnStateChangedEventArgs.cs:14]: @dreddy-work this needs to be raised as a breaking change in https://github.com/dotnet/docs/issues/22021

**Comment by dreddy-work** [comment]
Thanks @gpetrou  for taking time and getting this closure.

**Comment by dreddy-work** [comment]
tracking doc update here: https://github.com/dotnet/dotnet-api-docs/issues/7343

**Pull request merged**